lib/c-api/src/deprecated/import/wasi.rs

```diff
@@ -213,11 +213,6 @@ pub unsafe extern "C" fn wasmer_wasi_generate_default_import_object() -> *mut wa
     let mut wasi_state_builder = wasi::WasiState::new("wasmer-wasi-default-program-name");
     let wasi_state = wasi_state_builder.build().unwrap();
     let mut wasi_env = wasi::WasiEnv::new(wasi_state);
-    // this API will now leak a `Memory`
-    let memory_type = MemoryType::new(0, None, false);
-    let memory = Memory::new(store, memory_type).expect("create memory");
-    wasi_env.set_memory(memory);
-    // TODO(mark): review lifetime of `Memory` here
```

Comment:
I have no idea how this worked before. We should pass the memory exported from the instance. The previous code was not doing that :O

Reply:
I think it actually was, it's just that what this code was doing was being overwritten later.  It was really bad, glad it's being removed